packages/lib/server/repository/selectedCalendar.ts

```diff
@@ -435,4 +435,39 @@ export class SelectedCalendarRepository {
       unwatchAttempts: 0,
     });
   }
+
+  static async getNextBatchForSqlCache(limit = 100) {
+    // Get IDs of selected calendars that already have subscriptions
+    const existingSubscriptionIds = await prisma.calendarSubscription.findMany({
+      select: { selectedCalendarId: true },
+    });
+    const excludeIds = existingSubscriptionIds.map((sub) => sub.selectedCalendarId);
```

Comment:
DevinAI, this is very resource wasteful. This table is projected to have thousands of records. We care for selectedCalendars where calendarSubscriptionId is null